js/src/sdk/base.toolset.ts

```diff
@@ -46,12 +65,17 @@ export class ComposioToolSet {
     schema?: TSchemaProcessor;
   } = {};
 
-  constructor(
-    apiKey: string | null = null,
-    baseUrl: string | null = COMPOSIO_BASE_URL,
-    runtime: string | null = null,
-    _entityId: string = "default"
-  ) {
+  constructor({
+    apiKey,
+    baseUrl,
+    runtime,
+    entityId,
+  }: {
+    apiKey?: string | null;
+    baseUrl?: string | null;
+    runtime?: string | null;
+    entityId?: string;
+  }) {
```

Comment:
The constructor refactoring is good, but consider adding JSDoc documentation for the constructor parameters to improve code maintainability. For example:

```typescript
/**
 * Creates a new instance of ComposioToolSet
 * @param {Object} config - Configuration object
 * @param {string|null} config.apiKey - API key for authentication
 * @param {string|null} config.baseUrl - Base URL for API requests
 * @param {string|null} config.runtime - Runtime environment
 * @param {string} config.entityId - Entity ID for operations
 */
constructor({
  apiKey,
  baseUrl,
  runtime,
  entityId,
}: {
  apiKey?: string | null;
  baseUrl?: string | null;
  runtime?: string | null;
  entityId?: string;
})
```